components/DebugImage.tsx

```diff
@@ -23,25 +23,28 @@ export default function DebugImage(props) {
   const { error, message } = data;
   const debug = error || message;
 
+  const fetchData = useCallback(async () => {
+    const params = { w: props.width, h: props.height, url: props.src };
+    const result: any = await axios.get("/api/imgdim", { params });
+    const { optimized, error, message } = result.data;
+    if (optimized) {
+      setData({ error: null, message: null });
+    } else {
+      setData({ error, message });
+    }
+  }, [props.height, props.src, props.width]);
+
+  useEffect(() => {
+    if (!isDebugging(props)) {
+      fetchData();
+    }
+  }, [fetchData, props]);
+
```

Comment:
It looked like we didn't want to fetch data if we were debugging, so I've moved that logic into the hook. React always wants hooks to run if they're declared.